Find a functionally equivalent code:android.animation.ObjectAnimator.setTarget:COMMENT
Method Modifier: public      
Comment:/**
 * Sets the target object whose property will be animated by this animation
 *
 * @param target The object being animated
 */

@@ -1,11 +1,8 @@
 {
-    if (mTarget != target) {
-        final Object oldTarget = mTarget;
-        mTarget = target;
-        if (oldTarget != null && target != null && oldTarget.getClass() == target.getClass()) {
-            return;
-        }
-        // New target type should cause re-initialization prior to starting
+    final Object oldTarget = getTarget();
+    if (oldTarget != target) {
+        mTarget = target == null ? null : new WeakReference<Object>(target);
+        // New target should cause re-initialization prior to starting
         mInitialized = false;
     }
 }

